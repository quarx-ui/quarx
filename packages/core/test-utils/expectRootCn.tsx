import React, { FC } from 'react';
import { render } from '@testing-library/react';

export const expectRootCn = <T extends FC<{ className?: string, children: any }>>(Component: T, name: string) => {
    const className = 'test-class-name';
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    const { container: { firstChild: rootNode } } = render(<Component className={className} />);
    expect(rootNode).toHaveClass(className);
    expect(rootNode).toHaveClass(name);
};
