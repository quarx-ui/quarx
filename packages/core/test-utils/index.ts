export { expectRootCn } from './expectRootCn';
export * from './testBEM';
