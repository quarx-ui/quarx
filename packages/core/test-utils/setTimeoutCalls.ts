interface FakeSetTimeout {
    mock: {
        calls: Array<[() => void]>; // TODO Доработать типизацию
    };
}

/** При рендере jest использует setTimeout(_flushCallback, 0),
 * из-за чего проверка количества вызовов setTimeout происходит нестабильно.
 *
 * @return массив вызовов setTimeout без _flushCallback */
export const setTimeoutCalls = () => (setTimeout as unknown as FakeSetTimeout)
    .mock.calls
    .filter(([fn]) => !String(fn).includes('_flushCallback'));
