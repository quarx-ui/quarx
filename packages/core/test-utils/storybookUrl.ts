type PropsType = Record<string, string | boolean | number>;

interface UrlOptions {
    component: string;
    story?: string;
    props?: PropsType;
}

const reducePropsToURLString = (props?: PropsType) => {
    if (!props) {
        return '';
    }

    const keysOfProps = Object.keys(props);
    const valuesOfProps = Object.values(props);

    if (keysOfProps.length > 1) {
        return keysOfProps.reduce((prev, cur) => {
            const currentProp = `${cur}:${props[cur]}`;
            return `${prev};${currentProp}`;
        });
    }
    return `${keysOfProps[0]}:${valuesOfProps[0]}`;
};

export function storyUrl(options: UrlOptions) {
    const { component, story = 'sandbox', props } = options;
    const reducedProps = reducePropsToURLString(props);

    return `?path=/story/core-${component}--${story}&args=${reducedProps}`;
}
