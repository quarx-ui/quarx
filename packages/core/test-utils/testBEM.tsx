import React, { ComponentType } from 'react';
import { screen, render, cleanup } from '@testing-library/react';

export const expectPropsInClasses = (
    element: HTMLElement,
    qxClassname?: string,
) => (
    prop: string,
    value?: string | boolean,
) => {
    const qxSelector = qxClassname ?? element.classList[0];

    const valueIsBoolean = typeof value === 'boolean' || value === 'true' || value === 'false';
    const booleanClassname = value === true || value === 'true' ? `${qxSelector}_${prop}` : undefined;

    const expectedClassname = booleanClassname
        ?? (!valueIsBoolean
            ? `${qxSelector}_${prop}_${value}`
            : undefined);

    if (!expectedClassname) {
        return null;
    }

    return expect(element).toHaveClass(expectedClassname);
};

export const expectPropsMapInClasses = (element: HTMLElement, qxClassname?: string) => {
    const expectPropInClasses = expectPropsInClasses(element, qxClassname);

    return (expectedProps: Record<string, string | boolean>) => {
        Object.keys(expectedProps).forEach((prop) => {
            expectPropInClasses(prop, expectedProps[prop]);
        });
    };
};

export const testStyleParams = <
    S extends Record<string, any>,
    P extends Record<string, any> = Record<string, any>,
>(
    Component: ComponentType<P>,
    expectedDefaultParams: S,
    defaultProps: P = {} as P,
    qxClassname?: string,
) => (props: { [key in keyof S]: S[key][] }) => {
    describe('style params', () => {
        describe('default values', () => {
            const testId = 'testStyleParams-default';

            render(
                <Component
                    {...defaultProps}
                    data-testid={testId}
                />,
            );

            const element = screen.getByTestId(testId);

            Object.entries(expectedDefaultParams).forEach(([key, value]) => {
                it(`${key}`, () => {
                    expectPropsInClasses(element, qxClassname)(key, value);
                });
            });

            cleanup();
        });

        Object.entries(props).forEach(([propName, values]) => {
            describe(propName, () => {
                Object.values(values).forEach((value) => {
                    it(`${value}`, () => {
                        const testId = `testStyleParams-${propName}-${value}`;

                        const testProps = { [propName]: value };
                        const { asFragment } = render(
                            <Component
                                {...(defaultProps ?? {})}
                                {...testProps as P}
                                data-testid={testId}
                            />,
                        );

                        const element = screen.getByTestId(testId);
                        expectPropsMapInClasses(element, qxClassname)(testProps as S);
                        expect(asFragment()).toMatchSnapshot();
                    });
                });
            });
        });
    });
};
