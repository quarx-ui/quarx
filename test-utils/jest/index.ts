export { testRootCn } from './testRootCn';
export * from './testBEM';
