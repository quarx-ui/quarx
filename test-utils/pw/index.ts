export {
    compareStorySnapshots,
    compareVariantSnapshots,
    comparePropertySnapshots,
    comparePropertiesMapSnapshots,
} from './compareSnapshots';
