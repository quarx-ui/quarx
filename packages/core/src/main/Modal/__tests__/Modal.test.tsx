import React, { FC, useState } from 'react';
import { render, screen } from '@testing-library/react';
import { testStyleParams } from '@core/test-utils';
import { Button, Modal, ModalProps, ModalStyleParams } from '@core';
import userEvent from '@testing-library/user-event';

const buttonTestId = 'buttonTestId';

const Template: FC<ModalProps> = (props) => {
    const [isOpened, setIsOpened] = useState(false);

    return (
        <div>
            <Button
                onClick={() => setIsOpened(true)}
                data-testid={buttonTestId}
            />
            <Modal
                open={isOpened}
                {...props}
            />
        </div>
    );
};

describe('Modal', () => {
    testStyleParams<Partial<ModalStyleParams>, ModalProps>(
        Modal,
        {
            size: 'medium',
            scrollBehaviour: 'window',
        },
        { open: true },
        { capture: 'body', qxClassname: 'QxModal' },
    )({
        size: ['small', 'medium'],
        scrollBehaviour: ['body', 'window'],
    });

    it('modal in the document', async () => {
        const { asFragment } = render(
            <Template
                title="Title"
                subTitle="SubTitle"
                body="Body"
                footerButtons={{
                    success: {
                        children: 'Success',
                    },
                    secondary: {
                        children: 'Secondary',
                    },
                    danger: {
                        children: 'Danger',
                    },
                }}
            />,
        );

        const button = screen.getByTestId(buttonTestId);

        expect(asFragment().ownerDocument.body).toMatchSnapshot();
        userEvent.click(button);
        expect(asFragment().ownerDocument.body).toMatchSnapshot();
    });

    it('keepMounted modal always in the document', async () => {
        const { asFragment } = render(<Template
            title="Title"
            subTitle="SubTitle"
            body="Body"
            OverScreenProps={{
                keepMounted: true,
            }}
        />);

        const button = screen.getByTestId(buttonTestId);

        expect(asFragment().ownerDocument.body).toMatchSnapshot();
        userEvent.click(button);
        expect(asFragment().ownerDocument.body).toMatchSnapshot();
    });
});
