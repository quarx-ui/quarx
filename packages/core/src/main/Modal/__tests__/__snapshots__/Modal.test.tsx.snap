// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal keepMounted modal always in the document 1`] = `
<body
  style=""
>
  <div>
    <div>
      <button
        class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-xf29is-QxBaseButton-root-QxButton-root"
        data-testid="buttonTestId"
        type="button"
      />
    </div>
  </div>
  <div
    aria-hidden="true"
    class="QxOverScreen QxOverScreen_placement_center QxModal QxModal_size_medium QxModal_scrollBehaviour_window QxModal_hasHeader css-ykwgej-QxOverScreen-root-QxModal-root"
    data-component="QxOverScreen"
    data-index="0"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="true"
      class="QxBackdrop css-15lu5nv-QxBackdrop-root-QxBackdrop-root"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer css-1elz5fp-QxModal-scrollContainer-QxModal-scrollContainer"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box css-1qs5fkk-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_medium css-a0et6t-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-1lmvl89-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-1bxoecb-QxHeaderBlock-title"
            >
              Title
            </div>
            <div
              class="QxHeaderBlock-subTitle css-1sju9ey-QxHeaderBlock-subTitle"
            >
              SubTitle
            </div>
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-o1d20x-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
        <div
          class="QxModal-body css-l3ia4v-QxModal-body"
        >
          Body
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal keepMounted modal always in the document 2`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div>
    <div>
      <button
        class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-xf29is-QxBaseButton-root-QxButton-root"
        data-testid="buttonTestId"
        type="button"
      />
    </div>
  </div>
  <div
    aria-hidden="false"
    class="QxOverScreen QxOverScreen_placement_center QxOverScreen_mounted QxModal QxModal_size_medium QxModal_scrollBehaviour_window QxModal_hasHeader css-q5558t-QxOverScreen-root-QxModal-root"
    data-component="QxOverScreen"
    data-index="0"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="false"
      class="QxBackdrop css-kx5t24-QxBackdrop-root-QxBackdrop-root enter-active"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer css-edvtbv-QxModal-scrollContainer-QxModal-scrollContainer enter-active"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box css-1qs5fkk-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_medium css-a0et6t-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-1lmvl89-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-1bxoecb-QxHeaderBlock-title"
            >
              Title
            </div>
            <div
              class="QxHeaderBlock-subTitle css-1sju9ey-QxHeaderBlock-subTitle"
            >
              SubTitle
            </div>
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-o1d20x-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
        <div
          class="QxModal-body css-l3ia4v-QxModal-body"
          style="min-height: NaNpx;"
        >
          Body
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal modal in the document 1`] = `
<body
  style=""
>
  <div>
    <div>
      <button
        class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-xf29is-QxBaseButton-root-QxButton-root"
        data-testid="buttonTestId"
        type="button"
      />
    </div>
  </div>
</body>
`;

exports[`Modal modal in the document 2`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div>
    <div>
      <button
        class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-xf29is-QxBaseButton-root-QxButton-root"
        data-testid="buttonTestId"
        type="button"
      />
    </div>
  </div>
  <div
    aria-hidden="false"
    class="QxOverScreen QxOverScreen_placement_center QxOverScreen_mounted QxModal QxModal_size_medium QxModal_scrollBehaviour_window QxModal_hasHeader QxModal_hasFooter css-q5558t-QxOverScreen-root-QxModal-root"
    data-component="QxOverScreen"
    data-index="0"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="false"
      class="QxBackdrop css-kx5t24-QxBackdrop-root-QxBackdrop-root appear-active"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer css-edvtbv-QxModal-scrollContainer-QxModal-scrollContainer appear-active"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box css-1qs5fkk-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_medium css-a0et6t-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-1lmvl89-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-1bxoecb-QxHeaderBlock-title"
            >
              Title
            </div>
            <div
              class="QxHeaderBlock-subTitle css-1sju9ey-QxHeaderBlock-subTitle"
            >
              SubTitle
            </div>
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-o1d20x-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
        <div
          class="QxModal-body css-nwhtp5-QxModal-body"
          style="min-height: NaNpx;"
        >
          Body
        </div>
        <div
          class="QxFooterBlock QxFooterBlock_size_medium QxFooterBlock_direction_vertical css-jysxse-QxFooterBlock-root"
        >
          <button
            class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_brand QxBaseButton_type_contained QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-xf29is-QxBaseButton-root-QxButton-root"
            type="button"
          >
            <div
              class="QxButton-content css-m3eosc-QxButton-content"
            >
              Success
            </div>
          </button>
          <button
            class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_secondary QxBaseButton_type_outlined QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-r672q1-QxBaseButton-root-QxButton-root"
            type="button"
          >
            <div
              class="QxButton-content css-m3eosc-QxButton-content"
            >
              Secondary
            </div>
          </button>
          <button
            class="QxBaseButton QxBaseButton_size_medium QxBaseButton_color_danger QxBaseButton_type_outlined QxBaseButton_borderRadius_medium QxButton QxButton_size_medium css-1ooohn3-QxBaseButton-root-QxButton-root"
            type="button"
          >
            <div
              class="QxButton-content css-m3eosc-QxButton-content"
            >
              Danger
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal style params scrollBehaviour body 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div />
  <div
    aria-hidden="false"
    class="QxOverScreen QxOverScreen_placement_center QxOverScreen_mounted QxModal QxModal_size_medium QxModal_scrollBehaviour_body QxModal_hasHeader css-q5558t-QxOverScreen-root-QxModal-root"
    data-component="QxOverScreen"
    data-index="0"
    data-testid="testStyleParams-scrollBehaviour-body"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="false"
      class="QxBackdrop css-kx5t24-QxBackdrop-root-QxBackdrop-root appear-active"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer css-zpb0mb-QxModal-scrollContainer-QxModal-scrollContainer appear-active"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box css-3f9y17-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_medium css-a0et6t-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-1lmvl89-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-1bxoecb-QxHeaderBlock-title"
            />
            <div
              class="QxHeaderBlock-subTitle css-1sju9ey-QxHeaderBlock-subTitle"
            />
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-o1d20x-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal style params scrollBehaviour window 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div />
  <div
    aria-hidden="false"
    class="QxOverScreen QxOverScreen_placement_center QxOverScreen_mounted QxModal QxModal_size_medium QxModal_scrollBehaviour_window QxModal_hasHeader css-q5558t-QxOverScreen-root-QxModal-root"
    data-component="QxOverScreen"
    data-index="0"
    data-testid="testStyleParams-scrollBehaviour-window"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="false"
      class="QxBackdrop css-kx5t24-QxBackdrop-root-QxBackdrop-root appear-active"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer css-edvtbv-QxModal-scrollContainer-QxModal-scrollContainer appear-active"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box css-1qs5fkk-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_medium css-a0et6t-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-1lmvl89-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-1bxoecb-QxHeaderBlock-title"
            />
            <div
              class="QxHeaderBlock-subTitle css-1sju9ey-QxHeaderBlock-subTitle"
            />
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-o1d20x-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal style params size medium 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div />
  <div
    aria-hidden="false"
    class="QxOverScreen QxOverScreen_placement_center QxOverScreen_mounted QxModal QxModal_size_medium QxModal_scrollBehaviour_window QxModal_hasHeader css-q5558t-QxOverScreen-root-QxModal-root"
    data-component="QxOverScreen"
    data-index="0"
    data-testid="testStyleParams-size-medium"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="false"
      class="QxBackdrop css-kx5t24-QxBackdrop-root-QxBackdrop-root appear-active"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer css-edvtbv-QxModal-scrollContainer-QxModal-scrollContainer appear-active"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box css-1qs5fkk-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_medium css-a0et6t-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-1lmvl89-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-1bxoecb-QxHeaderBlock-title"
            />
            <div
              class="QxHeaderBlock-subTitle css-1sju9ey-QxHeaderBlock-subTitle"
            />
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-o1d20x-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Modal style params size small 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div />
  <div
    aria-hidden="false"
    class="QxOverScreen QxOverScreen_placement_center QxOverScreen_mounted QxModal QxModal_size_small QxModal_scrollBehaviour_window QxModal_hasHeader css-q5558t-QxOverScreen-root-QxModal-root"
    data-component="QxOverScreen"
    data-index="0"
    data-testid="testStyleParams-size-small"
    datatype="overscreen"
    role="presentation"
    tabindex="0"
  >
    <div
      aria-hidden="false"
      class="QxBackdrop css-kx5t24-QxBackdrop-root-QxBackdrop-root appear-active"
      data-component="QxBackdrop"
    />
    <div
      class="QxModal-scrollContainer css-edvtbv-QxModal-scrollContainer-QxModal-scrollContainer appear-active"
      data-component="QxOverScreen-child"
    >
      <div
        class="QxModal-box css-1i6u4nz-QxModal-box"
      >
        <div
          class="QxHeaderBlock QxHeaderBlock_size_small css-130u650-QxHeaderBlock-root"
        >
          <div
            class="QxHeaderBlock-content css-1lmvl89-QxHeaderBlock-content"
          >
            <div
              class="QxHeaderBlock-title css-152ga7g-QxHeaderBlock-title"
            />
            <div
              class="QxHeaderBlock-subTitle css-fm9gst-QxHeaderBlock-subTitle"
            />
          </div>
          <button
            class="QxBaseButton QxBaseButton_size_small QxBaseButton_color_secondary QxBaseButton_type_text QxBaseButton_borderRadius_medium QxIconButton QxIconButton_size_small QxHeaderBlock-closeButton css-o1d20x-QxBaseButton-root-QxIconButton-root-QxHeaderBlock-closeButton"
            type="button"
          >
            <div
              class="QxIconButton-icon css-5rjpdh-QxIconButton-icon"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.70714 17.7071C7.31661 18.0976 6.68345 18.0976 6.29292 17.7071C5.9024 17.3166 5.9024 16.6834 6.29292 16.2929L10.5858 12L6.29293 7.70711C5.9024 7.31658 5.9024 6.68342 6.29293 6.29289C6.68345 5.90237 7.31662 5.90237 7.70714 6.29289L12 10.5858L16.2929 6.29289C16.6834 5.90237 17.3166 5.90237 17.7071 6.29289C18.0977 6.68342 18.0977 7.31658 17.7071 7.70711L13.4142 12L17.7071 16.2929C18.0977 16.6834 18.0977 17.3166 17.7071 17.7071C17.3166 18.0976 16.6834 18.0976 16.2929 17.7071L12 13.4142L7.70714 17.7071Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
